nest/04.finest.md

```diff
@@ -0,0 +1,192 @@
+# 4 finest
+As explained throughout this getting-started, `finest` is the command targeted towards experienced users.
+
+## 4.1 Key concepts
+Before using `finest`, you have to understand how it works on a deeper level.
+`finest` relies on 4 key concepts:
+* The cache
+* Transactions
+* The dependency graph
+* Groups
+
+### 4.1.1 The cache
+[//]: # (TODO: add link to the section 3.2)
+The cache was explained in [the section 3.2]().
+
+### 4.1.2 Transactions
+A transaction is an *action of `finest` on the filesystem*.
+A transaction can be, for example, removing or upgrading a package.
+
+### 4.1.3 The dependency graph
+The dependency graph is a directed graph representing the dependencies of packages.
+In this graph, each node is either **a group of packages**, either **a single package**.
+Each node has a requirement that is fulfilled by another node.
+A requirement can be fulfilled by **one** and only **one** package.
+A requirement is noted as `repository::category/package#[version|version_requirement]`.
+
+The dependency graph contains a root node named `@root`.
+Using `finest`, you can add or remove requirements from any node you want.
+You can't add a node without it having a parent node, so it will be `@root` by default.
+
+When you're done making your transactions with `finest`, you have to solve the dependency graph.
+`finest` has applied those transactions on a copy of your dependency graph, and when you solve it, it's actually *merging* that copy with the graph currently used by the OS.
+
+### 4.1.4 Groups
+A group is basically a group of packages.
+Groups are a great way to manage very easily packages used for the same purpose.
+For example, if you're working on a school project, you can create a group containing all the packages you need for this purpose.
+Once your project is over, you can safely delete this group, which will remove all the packages inside it, on the condition that those packages only exist in this group, and not any other one.
+
+> Note
+>
+> Group names always start with the character `@`.
+
+### 4.1.5 Example of dependency graph transactions
+
+Let's say you want to install the package `dash` through the command line, without specifying any version.
+It will be resolved as the requirement `stable::sys-devel/dash#*`.
+You dependency graph will look like this:
+
+```
+@root
+├─ stable::sys-devel/dash#*
+├─ ...
+└─ ...
+```
+
+`finest` will try to find a way to fulfill this requirement.
+Since no version was specified, it will find the latest version available, let's say 0.5.10.
+So you'll find a new node in your graph:
+
+```
+@root
+├─ stable::sys-devel/dash#*
+|   └─ fulfilled by stable::sys-devel/dash#0.5.10
+└─ ...
+```
+
+Now, for the sake of the example, let's say this version of `dash` depends on any version of the package `glibc`, which has a major version superior or equal to 1.
+It's going to add a new requirement to the graph, which will be fulfilled by, for example, `stable::sys-devel/glibc#1.8.1`.
```

Comment:
```suggestion
It's going to add a new requirement to the graph, which will be fulfilled by, for example, `stable::sys-libs/glibc#1.8.1`.
```

(You should also fix the other occurrences of glibc)

Reply:
is it `sys-libs` or `sys-lib`? I found no `sys-libs` on stable.raven-os.org